plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -337,6 +337,7 @@ public function register_checkout_field( $options ) {
 				wc_get_logger()->warning(
 					sprintf( 'Registering select fields as optional is not supported. "%s" will be registered as required.', esc_html( $id ) )
 				);
+				return;
```

Comment:
We should not return here, we should change `required` value and still register it I think.